src/renderer/ThemeBackend.js

```diff
@@ -0,0 +1,225 @@
+var Color = require('color')
+
+function changeContrast (colorString, factor) {
+  // TODO make the black check code work
+  const color = Color(colorString).hex() === '#000000' ? Color('#010101') : Color(colorString)
+  if (color.isDark()) {
+    console.log('dark')
+    return color.lighten(factor).rgb().string()
+  } else if (color.isLight()) {
+    console.log('light')
+    return color.darken(factor).rgb().string()
+  }
+}
+
+function invertColor (colorString) {
+  const color = Color(colorString)
+  return color.negate().rgb().string()
+}
+
+function changeSaturation (colorString, factor) {
+  const color = Color(colorString)
+  if (factor < 0) {
+    return color.desaturate(factor *= -1).rgb().string()
+  } else if (factor > 0) {
+    return color.saturate(factor).rgb().string()
+  } else {
+    return color.rgb().string()
+  }
+}
+
+function undefinedGuard (value, func) {
+  if (typeof value === 'undefined') return undefined
+  else return func(value)
```

Comment:
no need for the else, looks weird
Maybe even do `return typeof value === 'undefined' ? undefined : func(value)`